### ISSUE TITLE
Problem with rc2 image

### ISSUE THREAD

**Issue by ctindel** [elastic/kibana-docker#56]
I didn't have this issue at first, but I removed all my images and re-pulled everything, is it possible that something was re-built and re-pushed with a typo in the kibana.yml or something?  

docker logs -f sa-demo-apachelogs-kibana-build
{"type":"log","@timestamp":"2017-11-04T04:23:43Z","tags":["fatal"],"pid":1,"message":"\"xpack.monitoring.ui.container.elasticsearch.enabled\" setting was not applied. Check for spelling errors and ensure that expected plugins are installed and enabled."}

I don't even set XPACK_MONITORING_UI_CONTAINER_ELASTICSEARCH_ENABLED in my docker-compose file, the kibana portion looks like this:

```
  sa-demo-apachelogs-kibana-build:
    image: docker.elastic.co/kibana/kibana:6.0.0-rc2
    depends_on:
      - sa-demo-apachelogs-es-build
    container_name: sa-demo-apachelogs-kibana-build
    environment:
      XPACK_MONITORING_ENABLED: "false"
      SERVER_HOST: "0.0.0.0"
      ELASTICSEARCH_URL: http://es:9200
      ELASTICSEARCH_USERNAME: elastic
      ELASTICSEARCH_PASSWORD: changeme
      XPACK_SECURITY_SECURECOOKIES: "true"
    ports:
      - 5601:5601
    networks:
      sa-demo-apachelogs-build-net:
        aliases:
          - kibana
```

I've recreated it with a minimal docker-compose.yml here:

```
$ cat docker-compose.yml
version: '2'
services:
  kibana:
    image: docker.elastic.co/kibana/kibana:6.0.0-rc2
    container_name: kibana
    environment:
      XPACK_MONITORING_ENABLED: "false"
      SERVER_HOST: "0.0.0.0"
      ELASTICSEARCH_URL: http://es:9200
      ELASTICSEARCH_USERNAME: elastic
      ELASTICSEARCH_PASSWORD: changeme
      XPACK_SECURITY_SECURECOOKIES: "true"
    ports:
      - 5601:5601
$ docker-compose up
Creating network "tmp_default" with the default driver
Creating kibana
Attaching to kibana
kibana    | {"type":"log","@timestamp":"2017-11-04T04:52:17Z","tags":["fatal"],"pid":1,"message":"\"xpack.monitoring.ui.container.elasticsearch.enabled\" setting was not applied. Check for spelling errors and ensure that expected plugins are installed and enabled."}
kibana exited with code 64
```

**Comment by hub-cap** [comment]
does that kibana image come with xpack installed? that setting still exists in 6.0 as per the xpack monitoring docs. i know elasticsearch will abort if there is a configuration setting it doesn’t know about (which is generally because of typos or didn’t install the plugin registering those strings), but i’m not sure about how kibana behaves for settings that it doesn’t know about. just a guess wo being in front of the images tho.

**Comment by ctindel** [comment]
I don't know, what's weird is this was working fine until I did an rmi for all my images and a clean build, then it started happening.

**Comment by ctindel** [comment]
I mean yes, our images are supposed to have xpack installed.

**Comment by jarpy** [comment]
Ah, I see. On Friday, I did fix a serious bug and re-push the image. The extra, X-Pack specific configuration defaults were not being applied to the `x-pack` (and thus the default) flavoured image. As of 94ab3393129f7fd5d9be612868138d45b8f22341, they are, including `xpack.monitoring.ui.container.elasticsearch.enabled`.

In your deployment, you are disabling Monitoring. I guess in Kibana that means "don't load the Monitoring plugin _at all_". Thus, without the Monitoring plugin, Kibana does not know what to do with the `xpack.monitoring.ui.container.elasticsearch.enabled` setting, and that's fatal.

```
$ docker run --rm -it -e XPACK_MONITORING_ENABLED=false docker.elastic.co/kibana/kibana:6.0.0-rc2
  log   [21:06:28.986] [fatal] "xpack.monitoring.ui.container.elasticsearch.enabled" setting was not applied. Check for spelling errors and ensure that expected plugins are installed and enabled.
```

**Comment by ctindel** [comment]
@jarpy maybe kibana-docker needs to be the place to set that, but only after checking if monitoring env isn't disabled?

**Comment by jarpy** [comment]
I don't think so. This is a design decision in Kibana that means you need to set two settings instead of one to disable monitoring.

**Comment by ctindel** [comment]
@jarpy oh, what is the other setting I need to do to make the image work now with monitoring disabled?

**Comment by jarpy** [comment]
I mean that you have to remove `xpack.monitoring.ui.container.elasticsearch.enabled`. That's why I think the interface could be better. Having to remove that line means there are two "touch points" required to disable Monitoring.

**Comment by ctindel** [comment]
Anybody who wants to disable monitoring has to build a new image and overwrite the kibana.yml file? It essentially means that the environment variable for disabling monitoring is broken?

**Comment by jarpy** [comment]
You don't have to build an image, you could bind-mount the `kibana.yml`, which is really the best practice for config anyway, but yeah, it's not pretty.

**Comment by jarpy** [comment]
Though, TBH, I don't know why anyone would disable Monitoring but not Security.

**Comment by ctindel** [comment]
The demo environment needs security enabled for obvious reasons, it's accessible on the internet. But we don't need monitoring. 

Disabling monitoring is very common, it seems strange to break things that used to work and make it more difficult to disable.

**Comment by jarpy** [comment]
For sure. Strict option parsing is the change that causes the breakage. It looks like it was introduced in Kibana 6:

https://github.com/elastic/kibana/pull/12554

We'll have to do a workaround of some kind.

**Comment by ctindel** [comment]
@jarpy what if remove xpack.monitoring.ui.container.elasticsearch.enabled from the kibana.yml in the docker image and just tell people if they want monitoring on they need to set that environment variable (or add it to the kibana.yml) ?  That way enabling and monitoring is easy.

**Comment by tylersmalley** [comment]
I have created an issue for this here: https://github.com/elastic/kibana/issues/14967

**Comment by jarpy** [comment]
Closing in favour of elastic/kibana#14967.
